### PR TITLE
Make input values work for recursive assemblies

### PR DESCRIPTION
**Pull request by klzns**
**What problem is this solving?**

Make Input Values work on recursive assemblies.

**How should this be manually tested?**

1. Open https://breno--storecomponents.myvtex.com/custom-bell/p
2. Click on Customize
3. A modal will open, click on the button "Testing"
4. Type anything in the fields
5. Click "DONE"
6. Click on Customize again
7. Check if the fields are filled with the inputs of step 4

### CODE REVIEW

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionsGroup.test.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { render, fireEvent } from '@vtex/test-tools/react'
+import { render, fireEvent, within } from '@vtex/test-tools/react'
```
Comment (iaronaraujo) [inline react/components/ProductAssemblyOptions/ProductAssemblyOptionsGroup.test.tsx]: Just out of curiosity, what's this `within`?

File: react/components/ProductAssemblyContext/Group.tsx [hunk 2]
```diff
@@ -62,15 +62,13 @@ export const ProductAssemblyGroupContextProvider: FC<ProductAssemblyGroupContext
...
   }, {})
 
-  const initialState: AssemblyOptionGroupState = {
-    ...assemblyOption,
-    path: groupPath,
-    quantitySum,
-    optin: assemblyOption.required,
-    valuesOfInputValues,
-  }
+  assemblyOption.path = assemblyOption.path ?? groupPath
```
Comment (jgfidelis) [inline react/components/ProductAssemblyContext/Group.tsx]: can't we try with the initializer solution?

Comment (klzns) [reply react/components/ProductAssemblyContext/Group.tsx]: Change to use the initializer solution

File: react/components/ProductAssemblyOptions/ProductAssemblyOptionsGroup.test.tsx (hunk 1 above)
Comment (klzns) [reply react/components/ProductAssemblyOptions/ProductAssemblyOptionsGroup.test.tsx]: This: https://testing-library.com/docs/dom-testing-library/api-helpers#within-and-getqueriesforelement-apis

**Pull request merged**